### PR TITLE
added Hermes-Retry-Count header in Consumers

### PR DESCRIPTION
**Pull request by adamdubiel**
also changed behavior of Retry-After not to slow down message sending when used

### CODE REVIEW

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSender.java [hunk 1]
```diff
@@ -146,35 +146,34 @@ public void accept(MessageSendingResult result) {
...
 
-        private boolean attemptResending(MessageSendingResult result) {
-            long delay = extractRetryDelay(result);
-            if (!willExceedTtl(message, delay) && subscriptionAllowsResending(result)) {
-                retrySingleThreadExecutor.schedule(() -> retrySending(result),
-                        delay, TimeUnit.MILLISECONDS);
-                return true;
-            }
-            return false;
+        private boolean attemptResending(MessageSendingResult result, long retryDelay) {
```
Comment (chemicL) [inline hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSender.java]: now this method should read as `shouldAttemptResending`

File: docs/docs/user/subscribing.md [hunk 2]
```diff
@@ -106,12 +107,25 @@ failure the rescue team has one hour to fix the problem before any event will be
...
 
+### Retries counter
+
+Each message sent by Hermes using HTTP sender comes with an additional header: **Hermes-Retry-Count**. It contains a number
+of retries for this specific message done by this Consumer instance. 
+
+The number of retries is counted locally, meaning it can not be treated as a global counter of delivery attempts.
+This counter will reset when:
+
+* Consumer instances is shut down before committing the offset to Kafka and other instances attempts to deliver this
```
Comment (chemicL) [inline docs/docs/user/subscribing.md]: I guess both bullet points can start with lowercase

**Comment by chemicL** [comment]
:+1:

**Comment by druminski** [comment]
:+1:

**Pull request merged**